integr8ly/integreatly-operator
# INTLY-6750-Enable ops addon in Fuse Online and remove alerts/dashboards/servicemonitors created by RHMI for Fuse Online

# Description
Due to a bug in Fuse Online 7.6 (https://github.com/syndesisio/syndesis/issues/8289) we had to create a subset of the Fuse Online alerts and dashboards in the RHMI Operator, instead of just enabling the Fuse Online ops addon.

- Ensure the ops addon for the latest release of Fuse Online does not create alerts for a database when external databases are enabled
- Remove the Fuse Online alerts/dashboards created from the RHMI operator (https://github.com/integr8ly/integreatly-operator/tree/19ffc3907adc138a0ef76b687bb2e88a4a821933/templates/monitoring/fuseonline) & update the reconcile logic to remove the template creation
- Enable the Fuse Online ops addon (here https://github.com/integr8ly/integreatly-operator/blob/19ffc3907adc138a0ef76b687bb2e88a4a821933/pkg/products/fuse/reconciler.go#L342-L352)
- As part of upgrade, add removal logic to delete the old resources from the cluster
- Create a follow up issue to remove the removal logic


## JIRA
https://issues.redhat.com/browse/INTLY-6750

## Type of change




## Checklist

## Verification
- Checkout master and install integreatly on the cluster
- Wait for the integreatly install to finish
- Check that there aren't any owner references in the yaml of the resources created from the [linked templates](https://github.com/integr8ly/integreatly-operator/tree/19ffc3907adc138a0ef76b687bb2e88a4a821933/templates/monitoring/fuseonline)
- Stop the operator
- Checkout this branch and start the operator
- After a few minutes return to the resources and ensure that they now have owner references and don't have any integreatly labels ( apart from the owner reference, which references a `syndesis` named `integreatly`
